flyteplugins/go/tasks/plugins/k8s/ray/batchscheduler/plugins.go

```diff
@@ -0,0 +1,28 @@
+package batchscheduler
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/flyteorg/flyte/flyteidl/gen/pb-go/flyteidl/plugins"
+	schedulerConfig "github.com/flyteorg/flyte/flyteplugins/go/tasks/plugins/k8s/ray/batchscheduler/config"
+	"github.com/flyteorg/flyte/flyteplugins/go/tasks/plugins/k8s/ray/batchscheduler/scheduler/kubernetes"
+	"github.com/flyteorg/flyte/flyteplugins/go/tasks/plugins/k8s/ray/batchscheduler/scheduler/yunikorn"
+)
+
+type SchedulerPlugin interface {
+	GetSchedulerName() string
+	ParseJob(config *schedulerConfig.Config, metadata *metav1.ObjectMeta, workerGroupsSpec []*plugins.WorkerGroupSpec, pod *v1.PodSpec, primaryContainerIdx int) error
```

Comment:
Could we not use ray WorkerGroupSpec here? I think it's better to make it generic so we don't need to write another spark/dash scheduler plugin in the future.